### DISCUSSION TITLE
WARN: Use matchDepNames instead of matchPackageNames

### DISCUSSION THREAD

**Discussion by MetRonnie** [renovatebot/renovate#28753] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

The other day we suddenly got this warning on our [dependency dashboard](https://github.com/cylc/cylc-ui/issues/1014):

![image](https://github.com/renovatebot/renovate/assets/61982285/8b039034-c420-41ce-98ed-2e4c9a4bdad1)

I don't understand why. In the [docs for `matchPackageNames`](https://docs.renovatebot.com/configuration-options/#matchpackagenames), it says

> **Note**
> `matchPackageNames` will try matching `packageName` first and then fall back to matching `depName`. If the fallback is used, Renovate will log a warning, because the fallback will be removed in a future release. Use `matchDepNames` instead.

But in the [docs for `matchDepNames`](https://docs.renovatebot.com/configuration-options/#matchdepnames) it says

> **Warning**
> For advanced use only! Use at your own risk!

without really explaining what `matchDepNames` does.

What is the difference between `matchPackageNames` and `matchDepNames`?

Is it just that we should use `matchDepNames` for yarn only (see logs below)? 

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: No currentVersion or lockedVersion found for cypress-io/github-action
DEBUG: hostRules: no authentication for registry.npmjs.org
WARN: Use matchDepNames instead of matchPackageNames
{
  "packageRule": {
    "matchPackageNames": [
      "yarn"
    ],
    "automerge": true
  }
  "packageName": "@yarnpkg/cli"
  "depName": "yarn"
}

DEBUG: PackageFiles.add() - Package file saved for base branch
{
  "baseBranch": "master"
}

DEBUG: Package releases lookups complete

```

</details>

**Comment by rarkins** [comment, marked as answer]
`matchPackageNames` matches against `packageName`, while `matchDepNames` matches against `depName`. In your case you could switch to matchDepNames=yarn or set matchPackageNames=@yarnpkg/cli

**Reply by MetRonnie** [reply]
Thanks, that's sorted it 👍
